src/snowflake/snowpark/_internal/udf_utils.py

```
@@ -1276,6 +1281,22 @@ def create_python_udf_or_sp(
     )
     imports_in_sql = f"IMPORTS=({all_imports})" if all_imports else ""
     packages_in_sql = f"PACKAGES=({all_packages})" if all_packages else ""
+    artifact_repository_in_sql = (
+        f"ARTIFACT_REPOSITORY={artifact_repository}" if artifact_repository else ""
+    )
+    artifact_repository_packages_str = (
+        "','".join(artifact_repository_packages) if artifact_repository_packages else ""
+    )
```

We already do this during UDxF creation but can we throw an exception if users specify `artifact_repository_packages` but not a corresponding `artifact_repository`?